sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -0,0 +1,389 @@
+      statement.executeUpdate("create table TEST(ID INT, NAME VARCHAR(200))");
+    } catch (Exception e) {
+      LOGGER.warn("Can't create table BEAM, it probably already exists", e);
+    } finally {
+      statement.close();
+    }
+
+    statement = connection.createStatement();
+    statement.executeUpdate("delete from BEAM");
+    statement.close();
+
+    String[] scientists = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday",
+        "Newton", "Bohr", "Galilei", "Maxwell"};
+    for (int i = 1; i <= 1000; i++) {
+      int index = i % scientists.length;
+      PreparedStatement preparedStatement = connection.prepareStatement("insert into BEAM "
+          + "values (?,?)");
+      preparedStatement.setInt(1, i);
+      preparedStatement.setString(2, scientists[index]);
+      preparedStatement.executeUpdate();
+      preparedStatement.close();
+    }
+
+    connection.commit();
+    connection.close();
+  }
+
+  /**
+   * Example of {@link PCollection} element that a
+   * {@link org.apache.beam.sdk.io.jdbc.JdbcIO.RowMapper} can return.
+   */
+  public class JdbcDataRecord implements Serializable {
+
+    private String[] tableNames;
+    private String[] columnNames;
+    private Object[] columnValues;
+    private int[] columnTypes;
+
+    public JdbcDataRecord() {
+    }
+
+    public JdbcDataRecord(int size) {
+      this.tableNames = new String[size];
+      this.columnNames = new String[size];
+      this.columnValues = new Object[size];
+      this.columnTypes = new int[size];
+    }
+
+    public String[] getTableNames() {
+      return tableNames;
+    }
+
+    public void setTableNames(String[] tableName) {
+      this.tableNames = tableName;
+    }
+
+    public String[] getColumnNames() {
+      return columnNames;
+    }
+
+    public void setColumnNames(String[] columnNames) {
+      this.columnNames = columnNames;
+    }
+
+    public Object[] getColumnValues() {
+      return columnValues;
+    }
+
+    public void setColumnValues(Object[] columnValues) {
+      this.columnValues = columnValues;
+    }
+
+    public int[] getColumnTypes() {
+      return columnTypes;
+    }
+
+    public void setColumnTypes(int[] columnTypes) {
+      this.columnTypes = columnTypes;
+    }
+
+    public String toString() {
+      StringBuilder builder = new StringBuilder();
+      for (int i = 0; i < tableNames.length; i++) {
+        builder.append("Table: ").append(tableNames[i]).append(" | Column: ")
+            .append(columnNames[i]).append(" | Type: ").append(columnTypes[i])
+            .append(" | Value: ").append(columnValues[i]).append("\n");
+      }
+      return builder.toString();
+    }
+
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testRead() throws Exception {
+    TestPipeline pipeline = TestPipeline.create();
+
+    PCollection<JdbcDataRecord> output = pipeline.apply(
+        JdbcIO.read()
+            .withDataSource(dataSource)
+            .withQuery("select * from BEAM")
```

Comment:
Using "select *" would be poor practice in an actual pipeline because it will break if fields in the table are reordered; or it will start performing poorly if a bulky column is added to the database, etc. I think "select *" is generally considered a code smell in database code and should be effectively never used in production code.

It's better to supply a query that explicitly lists the columns you're reading.